Animations/Animation.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+var React = require("react/addons");
+var Arrival = require('Arrival');
+var Easings = require("./AnimationEasings.js");
+var ReactTransitionGroup = React.addons.TransitionGroup;
+
+//makeTransition FUNCTION: creates animation for wrapped component based on props of wrapper component
+  //@param displayName STRING: intended name of the wrapper animation component
+  //@param transitions OBJECT: properties that define the default animation properties for the animation component wrapper
+var makeTransition = function (displayName, transitions) {
+
+  // setAnimationDomNode FUNCTION: function to handle manipulating and setting a TransitionGroup in the DOM.
+    //@param mount STRING: the transition key you are looking to affect
+    //@param done: done attribute passed from the containing function
+  var setAnimationDomNode = function (mount, done) {
+       // This is called at the same time as componentDidMount() for components added to an existing TransitionGroup. It will block other animations from occurring until callback is called. It will not be called on the initial render of a TransitionGroup.
+      this.el = this.getDOMNode();
+      this.$el = $(this.el);
+
+      // requestAnimationFrame FUNCTION: Calls the specified function to update an animation before the next browser repaint. Defined in window.
+      requestAnimationFrame(function () {
+        this.$el.css(transitions[mount]);
+
+        requestAnimationFrame(function () {
+          this.$el.css(transitions[mount]);
```

Comment:
In the current implementation this will not access the proper properties on this inner call to transitions. The keys for the function before were eg 'enter' and 'enterActive', as well as 'leave' and 'leaveActive'